rotkehlchen/exchanges/bittrex.py

```diff
@@ -274,6 +276,16 @@ def _single_api_query(
         except requests.exceptions.ConnectionError as e:
             raise RemoteError(f'Bittrex API request failed due to {str(e)}')
 
+        # Error due to unsync system clock (only for non-public endpoints)
+        if not public_endpoint and response.status_code == HTTPStatus.UNAUTHORIZED:
```

Comment:
This logic should not be here but in the end of the this function here: https://github.com/rotki/rotki/blob/d2793c3ff60bc4c6adf95d8fd7ac16115414e154/rotkehlchen/exchanges/bittrex.py#L238

The reason is that function checks that the result is proper json.  Here you are are doing the json parsing and then in that functoin the json parsin will happen again, which is a waste. Also here you are not checking that the json parsing did not fail, which would cause an exception.

Reply:
I see :(. My first approach was implementing it where you are saying, but then I realised it required to handle the exception on each external request case (e.g. api keys, balances, history). Due to the "culprit" is `api_timestamp = str(ts_now_in_ms())` in `_single_api_query()` I thought that would be the more appropriate place for raising it.
However, I did not take into account the json parsing as you expect.